### ISSUE TITLE
SC 3.3.8 Redundant entry, term "step" is ambiguous

### ISSUE THREAD

**Issue by WilcoFiers** [w3c/wcag#1345]
I've raised this concern on the mailing list and on surveys several times. It is unclear if this word means "every action taken by a user" or "a high level breakdown of a process, each on its own page or view". WCAG 2 should be clear which of these two interpretations is the correct one.

**Comment by alastc** [comment]
Linking to [one of those previous threads](https://lists.w3.org/Archives/Public/w3c-wai-gl/2020JanMar/0284.html).

**Comment by Helixopp** [comment]
Websters dictionary defines "process" as:

1. A systematic series of actions directed to some end. 

2. Continuous action, operation, or series of changes taking place in a definite manner.
 
In these cases each action would be a step.

**Comment by Helixopp** [comment]
Draft Response: The ordinary English definition of "process" sufficiently identifies steps in a process as "a systematic series of actions directed to some end." It also provides a second definition as "a continuous action, operation, or series of changes taking place in a definite manner."

Both definitions sufficiently articulate that a step in a process is a systematic series of actions to achieve a specific outcome.

A process isn't confined to a single page, or view, it is constrained to the outcome.

**Comment by alastc** [comment]
Hi @Helixopp,

Based on the issue raised, it does appear that when transposed to website usage, there are a few possible levels we could be talking about for step:

1. Every action taken by the user, e.g. click in input, type first character, type second character.
2. Per-input, e.g. enter text into first input, enter text into second input.
3. Page/view level actions, e.g. fill in fields on first page.

I think we would all discount (1) as not appropriate, but do we need to provide a definition or guidance that steers towards (2) or (3)?

It matters because in the SC text talks about what is available on "subsequent steps", so where does one step finish and another start?

We did have quite a long discussion (e.g. [a previous thread](https://lists.w3.org/Archives/Public/w3c-wai-gl/2020JanMar/0284.html)), and decided **_not_** to define step. The general feeling was that the common sense version of step was sufficient.

So do we need to update the [understanding doc](https://w3c.github.io/wcag/understanding/redundant-entry.html)? 

How about adding this before the paragraph starting "Requiring people to recall information already..."
<hr>
The criterion is asking that, if you ask for the same information twice, it is available or auto-populated. Processes can use different types of steps, for example you might consider a page a step, or a section of a form within a page. In the context of this criterion, a separate step means that the original input asking for the information is not immediately available.

**Comment by rachaelbradley** [comment]
I created [pull request 1597](https://github.com/w3c/wcag/pull/1597) with Alastair's suggested text.. 

Updated Draft response based on Helixopp:

Draft Response: The ordinary English definition of "process" sufficiently identifies steps in a process as "a systematic series of actions directed to some end." It also provides a second definition as "a continuous action, operation, or series of changes taking place in a definite manner."

Both definitions sufficiently articulate that a step in a process is a systematic series of actions to achieve a specific outcome.

Under these definitions, a process isn't confined to a single page, or view, it is constrained to the outcome. 

We agree that additional clarification outside a definition would be helpful and have added a paragraph to the understanding to help with this.

**Comment by alastc** [comment]
Hi Wilco,

I think the outcome of the discussion on the 26th was to accept the PR which took Andrew/Michael's suggestion to remove step entirely, so the updated version is live:
https://w3c.github.io/wcag/guidelines/22/#redundant-entry

In light of that, I think we can close this issue, but please reopen if you disagree.
